roles/hosted_engine_setup/tasks/create_storage_domain.yml

```diff
@@ -41,11 +41,15 @@
       auth: "{{ ovirt_auth }}"
     register: datacenter_facts
   - name: Fetch Datacenter ID
-    set_fact: >-
-      datacenter_id={{ cluster_facts.ovirt_clusters|json_query("[?id=='" + cluster_id + "'].data_center.id")|first }}
```

Comment:
Watch out this gets you the first item in the list your solution gives you the last.
(not sure if it matters here, just note)